src/reporters/json.ts

```diff
@@ -167,8 +168,51 @@ export function formatNetworkFields(network: NetworkInfo) {
   };
 }
 
+function journeyInfo(
+  journey: OutputFields['journey'],
+  type: OutputFields['type'],
+  status: Payload['status']
+) {
+  if (!journey) {
+    return;
+  }
+  return {
+    name: journey.name,
+    id: journey.id,
+    status: type === 'journey/end' ? status : undefined,
+  };
+}
+
+function stepInfo(
+  step: OutputFields['step'],
+  type: OutputFields['type'],
+  status: Payload['status']
+) {
+  if (!step) {
+    return;
+  }
+  return {
+    name: step.name,
+    index: step.index,
+    status: type === 'step/end' ? status : undefined,
+  };
+}
+
 export default class JSONReporter extends BaseReporter {
   _registerListeners() {
+    /**
+     * report the number of journeys that exists on a suite which
+     * could be used for better sharding
+     */
+    this.runner.on('start', ({ numJourneys }) => {
+      this.writeJSON({
+        type: 'synthetics/metadata',
+        root_fields: {
+          num_journeys: numJourneys,
```

Comment:
I'm OK with this, but we'll need to confirm that heartbeat doesn't exist this event. It shouldn't because it should only index events inside a journey context, but it's something to be aware of. 

We should follow-up by adding explicit support to `synthetics/metadata` events to heartbeat

Reply:
Yes thats correct, heartbeat doesn't also support `journey/register` events if i remember correctly. Would it still pipe these events to the stdout?